sumo-api/sumo/base/colored_category.pyx

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# Copyright 2004-present Facebook. All Rights Reserved.
+
+import csv
+import numpy as np
+cimport numpy as np
+import os
+
+
+class ColoredCategory(object):
+    """ Class to generate distinct RGB values for different categories of a dataset."""
+
+    def __init__(self, csv_file):
+        """ Create an object that holds color information of categories in
+        <dataset_name> dataset.
+
+        Inputs:
+        csv_file (str) - filepath to an external csv file containing color mapping
+            information.
+
+        Exceptions:
+        IOError - raised if csv_file does not exist.
+
+        Note:
+        The CSV file must be in the the following format:
+        id,category,is_evaluated,color
+
+        example:
+        id,category,is_evaluated,color
+        0,empty,False,(152, 245, 255)
+        1,accordion,False,(30, 144, 255)
+        2,air_conditioner,True,(122, 197, 205)
+        3,amplifier,False,(110, 123, 139)
+        ...
+        """
+        if not os.path.exists(csv_file):
+            raise IOError("File {} does not exist.".format(csv_file))
+
+        with open(csv_file, "r", newline="") as f:
+            reader = csv.DictReader(f)
+            self._data = [{
+                "category": line["category"],
+                "is_evaluated": line["is_evaluated"],
+                "color": tuple(map(int, line["color"].split(";")))} for line in reader]
+
+        self._reverse_data = {
+            self._data[id]["category"]: self._data[id]["color"] for id, x in enumerate(self._data)
```

Comment:
This seems to make the assumption that id is sequential, which is not a requirement in the file and may not hold true in general (e.g., if someone decides to only handle a subset of the categories).  Suggest storing _data as a dict of dicts (e.g., data[id]["category"] = "accordian") instead of as a list of dicts.

Reply:
After discussion, we decided to leave the current data structure intact and just inform the use of this limitation.